repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -987,10 +987,13 @@ private AtlasEntityHeader mapVertexToAtlasEntityHeader(AtlasVertex entityVertex,
         ret.setCreateTime(new Date(GraphHelper.getCreatedTime(entityVertex)));
         ret.setUpdateTime(new Date(GraphHelper.getModifiedTime(entityVertex)));
 
-        List<AtlasTermAssignmentHeader> termAssignmentHeaders = mapAssignedTerms(entityVertex);
-        ret.setMeanings(termAssignmentHeaders);
-        ret.setMeaningNames(termAssignmentHeaders.stream().map(AtlasTermAssignmentHeader::getDisplayText).collect(Collectors.toList()));
-
+        if(RequestContext.get().includeMeanings()) {
```

Comment:
mapVertexToAtlasEntityHeader is used at a lot of places and a lot of APIs. This will break all those APIs